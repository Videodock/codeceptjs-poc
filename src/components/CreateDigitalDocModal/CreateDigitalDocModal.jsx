import React from 'react';
import { makeStyles } from '@material-ui/styles';
import { useTranslation } from 'react-i18next';
import Button from '@material-ui/core/Button';
import MenuItem from '@material-ui/core/MenuItem';
import DialogTitle from '@material-ui/core/DialogTitle';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import Typography from '@material-ui/core/Typography/Typography';
import Input from '../Input/Input';
import Form from '../Form/Form';
import styles from './CreateDigitalDocModal.styles';

const useStyles = makeStyles(styles);

export default function CreateDigitalDocModal ({ initialFormValues, onFormSubmit }) {
  const { t }   = useTranslation('startCampaign');
  const classes = useStyles();

  const licenses  = [{ label: 'Public', value: 'https://www0.cpdl.org/wiki/index.php/ChoralWiki:CPDL' }];
  const languages = ['en'];

  return (
    <div className={classes.root}>
      <DialogTitle className={classes.title} disableTypography >
        <Typography className={classes.header} variant="h2">
          {t('create_digitial_doc.select_own_file')}
        </Typography>
      </DialogTitle>
      <Form onSubmit={onFormSubmit} initialValues={{ ...initialFormValues }}>
        {({ values, handleChange, handleSubmit, handleBlur, setFieldValue, setFieldTouched, isSubmitting }) => (
          <div className={classes.container}>
            <DialogContent>
              <Input
                label={t('create_digitial_doc.score')}
                name="url"
                value={values.url}
                onChange={handleChange}
                onBlur={handleBlur}
              />
              <Input
                label={t('create_digitial_doc.thumbnail')}
                value={values.thumbnailUrl}
                name="thumbnailUrl"
                onChange={handleChange}
                onBlur={handleBlur}
              />
              <Input
                label={t('create_digitial_doc.score_title')}
                name="title"
                value={values.title}
                onChange={handleChange}
                onBlur={handleBlur}
              />
              <Input
                label={t('create_digitial_doc.editor')}
                name="creator"
                value={values.creator}
                onChange={handleChange}
                onBlur={handleBlur}
              />
              {/* <Input
                  label={t('create_digitial_doc.publisher')}
                  value={values.campaignTitle}
                  onChange={handleChange}
                  onBlur={handleBlur}
                /> */}
              <Input
                label={t('create_digitial_doc.copyright')}
                value={values.license}
                name="license"
                onChange={handleChange}
                onBlur={handleBlur}
                select
              >
                {licenses.map(option => (
                  <MenuItem key={option.label} value={option.value}>
                    {option.label}
                  </MenuItem>
                ))}
              </Input>
              <Input
                label={t('create_digitial_doc.language')}
                name="language"
                value={values.language}
                onChange={handleChange}
                onBlur={handleBlur}
                select
              >
                {languages.map(option => (
                  <MenuItem key={option} value={option}>
                    {option}
                  </MenuItem>
                ))}
              </Input>
              <Input
                label={t('create_digitial_doc.notes')}
                value={values.description}
                name="description"
                onChange={handleChange}
                onBlur={handleBlur}
                rows={3}
                multiline
              />

            </DialogContent>
            <DialogActions>
              <div className={classes.buttons}>
                <Button
                  variant="contained"
                  color="primary"
                  onClick={handleSubmit}
                >
                  {t('create_digitial_doc.upload')}
                </Button>
              </div>
            </DialogActions>
          </div>
        )}
      </Form>
    </div>
  );
}

CreateDigitalDocModal.propTypes = {};

