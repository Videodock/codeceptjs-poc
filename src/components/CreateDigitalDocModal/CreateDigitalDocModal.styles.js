import { createStyles } from '@material-ui/styles';

export default () => createStyles({
  title: {
    display       : 'flex',
    justifyContent: 'space-between',
  },
  container: {
    position: 'relative',
    margin  : '0 auto',
    width   : '100%',
  },
});
