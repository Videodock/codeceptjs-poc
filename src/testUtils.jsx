import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import { ThemeProvider } from '@material-ui/styles';
import { I18nextProvider } from 'react-i18next';
import { ApolloProvider } from '@apollo/client';
import { BrowserRouter as Router } from 'react-router-dom';
import 'url-search-params-polyfill';
import i18n from '../src/i18n';
import client from './graphql';
import theme from './theme';
import store from './redux';

const wrapper = ({ children }) => (
  <ApolloProvider client={client}>
    <Provider store={store}>
      <ThemeProvider theme={theme}>
        <I18nextProvider i18n={i18n}>
          <Router>
            {children}
          </Router>
        </I18nextProvider>
      </ThemeProvider>
    </Provider>
  </ApolloProvider>
);

const customRender = (ui, options) => render(ui, { wrapper, ...options });

export { customRender as render };
