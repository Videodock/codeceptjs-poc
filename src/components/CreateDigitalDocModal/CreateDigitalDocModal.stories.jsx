import React from 'react';
import { storiesOf } from '@storybook/react';
import CreateDigitalDocModal from './CreateDigitalDocModal';

storiesOf('CreateDigitalDocModal', module)
  .add('basic', () => <CreateDigitalDocModal />);
