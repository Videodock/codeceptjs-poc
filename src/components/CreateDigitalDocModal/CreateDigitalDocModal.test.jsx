import React from 'react';
import { render } from '../../testUtils';
import CreateDigitalDocModal from './CreateDigitalDocModal';

describe('<CreateDigitalDocModal />', () => {
  test('renders and matches snapshot', () => {
    const { container } = render(<CreateDigitalDocModal />);

    expect(container.firstChild).toMatchSnapshot();
  });

  // add your unit tests here
});
