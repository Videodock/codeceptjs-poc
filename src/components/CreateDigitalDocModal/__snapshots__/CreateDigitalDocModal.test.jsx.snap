// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateDigitalDocModal /> renders and matches snapshot 1`] = `
<div
  class="makeStyles-root-1"
>
  <div
    class="MuiDialogTitle-root makeStyles-title-2"
  >
    <h1
      class="MuiTypography-root makeStyles-header-3 MuiTypography-h1"
    >
      Select your own file
    </h1>
  </div>
  <form
    class="makeStyles-container-5"
  >
    <div
      class="MuiDialogContent-root"
    >
      <div
        class="makeStyles-root-39"
      >
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          Score
        </p>
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        >
          <div
            class="MuiInputBase-root MuiFilledInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiFilledInput-adornedEnd"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiFilledInput-input MuiInputBase-inputAdornedEnd MuiFilledInput-inputAdornedEnd"
              name="url"
              required=""
              type="url"
              value=""
            />
            <div
              class="MuiInputAdornment-root MuiInputAdornment-filled MuiInputAdornment-positionEnd"
            >
              <button
                class="MuiButtonBase-root MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="makeStyles-root-39"
      >
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          Thumbnail
        </p>
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        >
          <div
            class="MuiInputBase-root MuiFilledInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiFilledInput-adornedEnd"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiFilledInput-input MuiInputBase-inputAdornedEnd MuiFilledInput-inputAdornedEnd"
              name="thumbnailUrl"
              required=""
              type="text"
              value=""
            />
            <div
              class="MuiInputAdornment-root MuiInputAdornment-filled MuiInputAdornment-positionEnd"
            >
              <button
                class="MuiButtonBase-root MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        class="makeStyles-root-39"
      >
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          Score title
        </p>
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        >
          <div
            class="MuiInputBase-root MuiFilledInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiFilledInput-input"
              name="title"
              required=""
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="makeStyles-root-39"
      >
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          Editor
        </p>
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        >
          <div
            class="MuiInputBase-root MuiFilledInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiFilledInput-input"
              name="creator"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="makeStyles-root-39"
      >
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          Publisher
        </p>
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        >
          <div
            class="MuiInputBase-root MuiFilledInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiFilledInput-input"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        class="makeStyles-root-39"
      >
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          Copyright license
        </p>
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        >
          <div
            class="MuiInputBase-root MuiFilledInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
          >
            <div
              aria-haspopup="listbox"
              aria-labelledby=" mui-component-select-license"
              class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-filled MuiInputBase-input MuiFilledInput-input"
              id="mui-component-select-license"
              role="button"
              tabindex="0"
            >
              <span>
                ​
              </span>
            </div>
            <input
              name="license"
              type="hidden"
              value=""
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconFilled"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        class="makeStyles-root-39"
      >
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          Language
        </p>
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        >
          <div
            class="MuiInputBase-root MuiFilledInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
          >
            <div
              aria-haspopup="listbox"
              aria-labelledby=" mui-component-select-language"
              class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-filled MuiInputBase-input MuiFilledInput-input"
              id="mui-component-select-language"
              role="button"
              tabindex="0"
            >
              <span>
                ​
              </span>
            </div>
            <input
              name="language"
              type="hidden"
              value=""
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconFilled"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
        </div>
      </div>
      <div
        class="makeStyles-root-39"
      >
        <p
          class="MuiTypography-root MuiTypography-body1"
        >
          Miscellaneous Notes
        </p>
        <div
          class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        >
          <div
            class="MuiInputBase-root MuiFilledInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline MuiFilledInput-multiline"
          >
            <textarea
              aria-invalid="false"
              class="MuiInputBase-input MuiFilledInput-input MuiInputBase-inputMultiline MuiFilledInput-inputMultiline"
              name="description"
              required=""
              rows="3"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="MuiDialogActions-root MuiDialogActions-spacing"
    >
      <div>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
          tabindex="0"
          type="submit"
        >
          <span
            class="MuiButton-label"
          >
            Upload
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </form>
</div>
`;
